src/libcollections/string.rs

```diff
@@ -826,6 +826,38 @@ impl_eq! { String, &'a str }
 impl_eq! { Cow<'a, str>, String }
```

Comment:
Btw, you can use `&[..]` notation inside of `impl_eq!` like #23521 does and reduce all the impls for `Cow<'a, str>`, `String` and `str` to just a few lines.